cmake/traccc-compiler-options-cpp.cmake

```diff
@@ -22,8 +22,10 @@ if( ( "${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU" ) OR
    traccc_add_flag( CMAKE_CXX_FLAGS "-Wextra" )
    traccc_add_flag( CMAKE_CXX_FLAGS "-Wshadow" )
    traccc_add_flag( CMAKE_CXX_FLAGS "-Wunused-local-typedefs" )
-   traccc_add_flag( CMAKE_CXX_FLAGS "-pedantic" )
-   traccc_add_flag( CMAKE_CXX_FLAGS "-Wfloat-conversion" )
+   traccc_add_flag( CMAKE_CXX_FLAGS "-Wpedantic" )
+   if(PROJECT_IS_TOP_LEVEL)
```

Comment:
Where is `PROJECT_IS_TOP_LEVEL` meant to come from?

Reply:
It's built in as of CMake 3.21 (https://cmake.org/cmake/help/latest/variable/PROJECT_IS_TOP_LEVEL.html), and we already require 3.22 at minimum.

Reply:
Okay. I didn't use it so far, but since we need s pretty new CMake version for this code anyway...